tools/src/main/java/org/apache/kafka/tools/StreamsCleanupClient.java

```diff
@@ -0,0 +1,208 @@
+        try {
+            client.subscribe(topics);
+            client.poll(1);
+
+            for (TopicPartition partition : client.assignment()) {
+                client.seek(partition, 0);
+            }
+            client.commitSync();
+        } catch (RuntimeException e) {
+            System.err.println("Resetting source topic offsets failed");
+            throw e;
+        } finally {
+            client.close();
+        }
+    }
+
+    private void seekToEndIntermediateTopics(ZkUtils zkUtils, List<String> allTopics) {
+        if (options.has(intermediateTopicsOption)) {
+            System.out.println("Seek-to-end for intermediate user topics.");
+
+            final Properties config = new Properties();
+            config.setProperty(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, options.valueOf(bootstrapServerOption));
+            config.setProperty(ConsumerConfig.GROUP_ID_CONFIG, options.valueOf(applicationIdOption));
+            config.setProperty(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false");
+
+            for (String topic : options.valuesOf(intermediateTopicsOption)) {
+                if (allTopics.contains(topic)) {
+                    System.out.println("Topic: " + topic);
+
+                    final KafkaConsumer<byte[], byte[]> client = new KafkaConsumer<>(config, new ByteArrayDeserializer(), new ByteArrayDeserializer());
+                    try {
+                        client.subscribe(Collections.singleton(topic));
+                        client.poll(1);
+
+                        final Set<TopicPartition> partitions = client.assignment();
+                        client.seekToEnd(partitions);
+                        client.commitSync();
+                    } catch (RuntimeException e) {
+                        System.err.println("Seek to end for topic " + topic + " failed");
+                        throw e;
+                    } finally {
+                        client.close();
+                    }
+                }
+            }
+
+            System.out.println("Done.");
+        } else {
+            System.out.println("No intermediate topics specified.");
+        }
+    }
+
+    private void deleteInternalTopics(ZkUtils zkUtils, List<String> allTopics) {
+        System.out.println("Deleting internal topics.");
+
+        final String applicationId = options.valueOf(applicationIdOption);
+
+        for (String topic : allTopics) {
+            if (topic.startsWith(applicationId + "-") && (topic.endsWith("-changelog") || topic.endsWith("-repartition"))) {
+                TopicCommand.TopicCommandOptions commandOptions = new TopicCommand.TopicCommandOptions(new String[] {
+                    "--zookeeper", options.valueOf(zookeeperOption),
+                    "--delete", "--topic", topic});
+                try {
+                    TopicCommand.deleteTopic(zkUtils, commandOptions);
```

Comment:
I think we should also log which exact internal topics were being removed to prevent "Wait, what exactly did just happen?" questions.  Perhaps we can make this a CLI option, and we can discuss what the default is?